api/src/chat/repositories/block.repository.ts

```diff
@@ -89,16 +89,112 @@ export class BlockRepository extends BaseRepository<
       Block,
       'findOneAndUpdate'
     >,
-    _criteria: TFilterQuery<Block>,
+    criteria: TFilterQuery<Block>,
     _updates:
       | UpdateWithAggregationPipeline
       | UpdateQuery<Document<Block, any, any>>,
   ): Promise<void> {
     const updates: BlockUpdateDto = _updates?.['$set'];
+    if (updates?.category) {
+      const movedBlockId = criteria._id;
+
+      // Find and update blocks that reference the moved block
+      await this.model.updateMany(
+        { nextBlocks: movedBlockId },
+        { $pull: { nextBlocks: movedBlockId } },
+      );
+
+      await this.model.updateMany(
+        { attachedBlock: movedBlockId },
+        { $set: { attachedBlock: null } },
+      );
+    }
 
     this.checkDeprecatedAttachmentUrl(updates);
   }
 
+  /**
+   * Pre-processing logic for updating blocks.
+   *
+   * @param query - The query to update blocks.
+   * @param criteria - The filter criteria for the update query.
+   * @param updates - The update data.
+   */
+  async preUpdateMany(
+    _query: Query<
+      Document<Block, any, any>,
+      Document<Block, any, any>,
+      unknown,
+      Block,
+      'updateMany',
+      Record<string, never>
+    >,
+    _criteria: TFilterQuery<Block>,
+    _updates: UpdateQuery<Document<Block, any, any>>,
+  ): Promise<void> {
+    const ids: string[] = _criteria._id?.$in || [];
```

Comment:
I think we need to do a check on the existence of the variables before. (_criteria._id?.$in, _updates.$set.category, ...)